src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -0,0 +1,329 @@
+            if (q.getMaxResults() > 0) {
+                LOG.error("Deleting inspection hooks for inspectedPortId failed! inspectedPortId: " + portId);
+            }
+            return null;
+        });
+
+    }
+
+    // Inspection port methods
+    @Override
+    public InspectionPortElement getInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        if (inspectionPort == null) {
+            return null;
+        }
+
+        try {
+            String portId = inspectionPort.getElementId();
+
+            return this.txControl.required(() -> {
+                InspectionPortEntity ipEntity = this.utils.txInspectionPortEntityById(portId);
+                return ipEntity;
+            });
+        } catch (Exception e) {
+            LOG.warn("Failed to retrieve inspectionPort by id! Trying by ingress and egress.");
+        }
+
+        NetworkElement ingress = inspectionPort.getIngressPort();
+        NetworkElement egress = inspectionPort.getEgressPort();
+
+        InspectionPortEntity ipEntity = this.utils.findInspPortByNetworkElements(ingress, egress);
+        return ipEntity;
+    }
+
+    @Override
+    public Element registerInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        return this.txControl.required(() -> {
+
+            // must be within this transaction, because if the DB retrievals inside makeInspectionPortEntry
+            // are inside the required() call themselves. That makes them a part of a separate transaction
+
+            NetworkElement ingress = inspectionPort.getIngressPort();
+            NetworkElement egress = inspectionPort.getEgressPort();
+            InspectionPortEntity inspectionPortEntity = this.utils.findInspPortByNetworkElements(ingress, egress);
+
+            if (inspectionPortEntity == null) {
+                inspectionPortEntity = this.utils.makeInspectionPortEntity(inspectionPort);
+            }
+
+            inspectionPortEntity = this.em.merge(inspectionPortEntity);
+
+            return inspectionPortEntity;
+        });
+    }
+
+    @Override
+    public void setInspectionHookOrder(NetworkElement inspectedPort, InspectionPortElement inspectionPort, Long order)
+            throws Exception {
+
+        this.txControl.required(() -> {
+            InspectionHookEntity entity = this.utils.findInspHookByInspectedAndPort(inspectedPort, inspectionPort);
+            entity.setOrder(order);
+            return null;
+        });
+    }
+
+    @Override
+    public Long getInspectionHookOrder(NetworkElement inspectedPort, InspectionPortElement inspectionPort)
+            throws Exception {
+        InspectionHookElement inspectionHook = getInspectionHook(inspectedPort, inspectionPort);
+        return inspectionHook == null ? null : inspectionHook.getOrder();
+    }
+
+    @Override
+    public void updateInspectionHook(InspectionHookElement existingInspectionHook) throws Exception {
+
+        if (existingInspectionHook == null) {
```

Comment:
throw exception if you are getting nulls as arguments for cases where null is not expected